specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/costmanagement.benefits.json

```diff
@@ -0,0 +1,1002 @@
+      "properties": {
+        "properties": {
+          "type": "object",
+          "$ref": "#/definitions/benefitRecommendationProperties"
+        }
+      }
+    },
+    "benefitRecommendationProperties": {
+      "type": "object",
+      "description": "The properties of the benefit plan recommendation.",
+      "discriminator": "scope",
+      "properties": {
+        "firstConsumptionDate": {
+          "type": "string",
+          "description": "The first usage date used for looking back for computing the recommendation.",
+          "readOnly": true
+        },
+        "lastConsumptionDate": {
+          "type": "string",
+          "description": "The last usage date used for looking back for computing the recommendation.",
+          "readOnly": true
+        },
+        "lookBackPeriod": {
+          "type": "string",
+          "description": "The number of days of usage used to look back for computing the recommendation.",
+          "readOnly": true
+        },
+        "totalHours": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The total hours for which the cost is covered. Its equal to number of records in a property 'properties/usage/charges'.",
+          "readOnly": true
+        },
+        "usage": {
+          "type": "object",
+          "description": "The additional details about the usage which are considered while computing benefit recommendations.",
+          "$ref": "#/definitions/recommendationUsageDetails"
+        },
+        "armSkuName": {
+          "description": "ARM SKU name. For example, Standard_D1_V2 for Reservations and Compute_Savings_Plan for SavingsPlan.",
+          "type": "string",
+          "readOnly": true
+        },
+        "term": {
+          "type": "string",
+          "description": "Term period of the benefit. For example, P1Y or P3Y.",
+          "readOnly": true
+        },
+        "commitmentGranularity": {
+          "description": "Grain of the proposed commitment amount. For example, hourly, monthly etc.",
+          "$ref": "#/definitions/Grain"
+        },
+        "currencyCode": {
+          "type": "string",
+          "description": "An ISO 4217 currency code identifier for the costs and savings amounts.",
+          "readOnly": true
+        },
+        "costWithoutBenefit": {
+          "type": "number",
+          "format": "decimal",
+          "description": "The current cost without benefit corresponds to 'totalHours' property in a given look-back period.",
+          "readOnly": true
+        },
+        "recommendationDetails": {
+          "type": "object",
+          "description": "The details of the proposed recommendation.",
+          "$ref": "#/definitions/allSavingsBenefitDetails"
+        },
+        "allRecommendationDetails": {
+          "type": "object",
+          "description": "The list of all benefit recommendations.",
+          "readOnly": true,
+          "$ref": "#/definitions/allSavingsList"
```

Comment:
Whats the difference between 'proposed' and 'all'?

Reply:
This API is called by the Azure portal UX during the Savings Plan purchase workflow.
Our recommendation engine checks customer's usage in the last 7, 30, 60 days and recommends the commitment which will offer the best savings. That's given by 'proposed' recommendation.
But we provide all other recommendations generated by the engine in the 'all' recommendations object. Customers can choose any one out of this list or can enter custom amount.